patches/minecraft/net/minecraft/network/play/client/CPacketClickWindow.java.patch

```diff
@@ -1,11 +1,56 @@
 --- ../src-base/minecraft/net/minecraft/network/play/client/CPacketClickWindow.java
 +++ ../src-work/minecraft/net/minecraft/network/play/client/CPacketClickWindow.java
-@@ -55,7 +55,7 @@
+@@ -1,16 +1,21 @@
+ package net.minecraft.network.play.client;
+ 
+ import java.io.IOException;
++
++import net.minecraft.client.Minecraft;
++import net.minecraft.entity.player.EntityPlayer;
+ import net.minecraft.inventory.ClickType;
+ import net.minecraft.item.ItemStack;
+ import net.minecraft.network.Packet;
+ import net.minecraft.network.PacketBuffer;
+ import net.minecraft.network.play.INetHandlerPlayServer;
++import net.minecraftforge.common.util.PacketUtil;
+ import net.minecraftforge.fml.relauncher.Side;
+ import net.minecraftforge.fml.relauncher.SideOnly;
+ 
+ public class CPacketClickWindow implements Packet<INetHandlerPlayServer>
+ {
++    private EntityPlayer player;
+     private int field_149554_a;
+     private int field_149552_b;
+     private int field_149553_c;
+@@ -23,14 +28,15 @@
+     }
+ 
+     @SideOnly(Side.CLIENT)
+-    public CPacketClickWindow(int p_i46882_1_, int p_i46882_2_, int p_i46882_3_, ClickType p_i46882_4_, ItemStack p_i46882_5_, short p_i46882_6_)
++    public CPacketClickWindow(EntityPlayer player, int windowIdIn, int slotIdIn, int usedButtonIn, ClickType modeIn, ItemStack clickedItemIn, short actionNumberIn)
+     {
+-        this.field_149554_a = p_i46882_1_;
+-        this.field_149552_b = p_i46882_2_;
+-        this.field_149553_c = p_i46882_3_;
+-        this.field_149551_e = p_i46882_5_.func_77946_l();
+-        this.field_149550_d = p_i46882_6_;
+-        this.field_149549_f = p_i46882_4_;
++        this.player = player;
++        this.field_149554_a = windowIdIn;
++        this.field_149552_b = slotIdIn;
++        this.field_149553_c = usedButtonIn;
++        this.field_149551_e = clickedItemIn.func_77946_l();
++        this.field_149550_d = actionNumberIn;
++        this.field_149549_f = modeIn;
+     }
+ 
+     public void func_148833_a(INetHandlerPlayServer p_148833_1_)
+@@ -55,7 +61,7 @@
          p_148840_1_.writeByte(this.field_149553_c);
          p_148840_1_.writeShort(this.field_149550_d);
          p_148840_1_.func_179249_a(this.field_149549_f);
 -        p_148840_1_.func_150788_a(this.field_149551_e);
-+        net.minecraftforge.common.util.PacketUtil.writeItemStackFromClientToServer(p_148840_1_, this.field_149551_e);
++        PacketUtil.writeItemStackFromClientToServer(p_148840_1_, this.player, this.field_149551_e);
```

Comment:
This method is called on the netty network thread. Calling out to arbitrary methods in Item seems like a bad idea here.